### PR TITLE
synchronous retries + error logging in checkResponses

### PR DESCRIPTION
**Pull request by raahi-chada2004**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,11 +71,14 @@ Here is an example of an `OUTPUT` section:
...
 
 The `Exactly_Once` field takes in a boolean that describes whether exactly-once semantics will be utilized. By default, this field has value false and exactly-once is not used. With exactly-once delivery, response checking will be synchronous (as opposed to asynchronous response checking with at-least once/default semantics).
 
+The `Num_Synchronous_Retries` field takes in the maximum number of synchronous retries the plugin will attempt when streaming data with exactly once semantics. This field does not change the number of asynchronous retries attempted with at-least once/default semantics.
```
Comment (agrawal-siddharth) [inline README.md]: Please document the default value of this field.

File: out_writeapi.go [hunk 2]
```diff
@@ -313,6 +338,12 @@ func FLBPluginInit(plugin unsafe.Pointer) int {
 		log.Printf("Invalid Exactly_Once parameter in configuration file: %s", err)
 	}
 
+	//optional num synchronous retries parameter
```
Comment (agrawal-siddharth) [inline out_writeapi.go:341]: Please augment the comment to indicate this is used only for exactly-once.

File: retry.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package main
```
Comment (agrawal-siddharth) [inline retry.go:14]: Please add the boilerplate copyright header.

**Review by agrawal-siddharth — CHANGES_REQUESTED** [review]

File: retry.go (hunk 3 above)
Comment (raahi-chada2004) [reply retry.go:14]: Done.

File: out_writeapi.go (hunk 2 above)
Comment (raahi-chada2004) [reply out_writeapi.go:341]: Done.

File: README.md (hunk 1 above)
Comment (raahi-chada2004) [reply README.md]: Done.

**Review by agrawal-siddharth — APPROVED** [review]

**Pull request merged**